Framework/HistogramData/src/Histogram.cpp

```diff
@@ -253,5 +253,21 @@ template <> void Histogram::checkSize(const BinEdges &binEdges) const {
     throw std::logic_error("Histogram: size mismatch of BinEdges\n");
 }
 
+/** Resets the size of the internal x, dx, y, and e data structures
+
+  The argument refers to the requested y length. All other lengths are
+  then inferred automatically.*/
+void Histogram::resize(size_t n) {
+  auto newXSize = xMode() == XMode::Points ? n : n + 1;
+
+  m_x.access().mutableRawData().resize(newXSize);
+  m_y.access().mutableRawData().resize(n);
+  m_e.access().mutableRawData().resize(n);
+
+  if (m_dx.use_count() > 0) {
```

Comment:
We recently added `operator bool()` to `cow_ptr`, so you could simply write `if(m_dx)`, I think.